tests/add_accept_finalize_user_test.go

```diff
@@ -0,0 +1,147 @@
+package tests
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+
+	decision_models "github.com/HackIllinois/api/services/decision/models"
+	user_models "github.com/HackIllinois/api/services/user/models"
+	"github.com/dghubble/sling"
+	"gopkg.in/mgo.v2"
+)
+
+var admin_client *sling.Sling
+var mongo_session *mgo.Session
+
+func GetAdminClient() *sling.Sling {
+	// First, get an admin authorization token by running `make setup`.
+	path, err := os.Getwd()
+	if err != nil {
+		fmt.Printf("ERROR: %v\n", err)
+	}
+
+	cmd := exec.Command("make", "setup")
+	cmd.Dir = filepath.Dir(path)
+	out, err := cmd.Output()
+
+	if err != nil {
+		fmt.Printf("ERROR: %v\n", err)
+	}
+
+	out_lines := strings.Split(string(out[:]), "\n")
+	admin_token := out_lines[len(out_lines)-3]
+
+	return sling.New().Base("http://localhost:8000").Client(nil).Add("Authorization", admin_token)
+}
+
+func GetLocalMongoSession() *mgo.Session {
+	session, err := mgo.Dial("localhost")
+	if err != nil {
+		fmt.Println("Failed to connect to database:", err)
+		os.Exit(1)
+	}
+	return session
+}
+
+func TestMain(m *testing.M) {
+	admin_client = GetAdminClient()
+
+	mongo_session = GetLocalMongoSession()
+	mongo_session.DB("user").DropDatabase()
+	mongo_session.DB("decision").DropDatabase()
```

Comment:
Tests are still failing on subsequent runs after the first run.

I think that this cleanup is not being handled properly. We should be using the test db names in the config file (test-user, test-decision, etc.) instead of the dev db names. We can load the test config in `TestMain` and use `config.USER_DB_NAME` and `config.DECISION_DB_NAME` to specify the correct db names.

Reply:
The API is still being run using the development config (`make run`) since I did not switch to using the test config in this PR. So the correct databases are being dropped here. But I agree we should use the config instead.

Reply:
Ah, I see. I assumed that the API is being started by `scripts/run-test.sh `. But yes, we can change this in another PR.